exec/cnex/nstring.c

```diff
@@ -114,6 +114,28 @@ int string_compareString(TString *lhs, TString *rhs)
     return r;
 }
 
+TString *string_subString(TString *s, size_t startAt, size_t count)
+{
+    TString *r = string_newString();
+
+    size_t start = startAt;
+    size_t end = start + count;
+
+    if (end > s->length) {
+        end = s->length;
+    }
+
+    if (start <= 0 || start >= s->length) {
+        start = 0;
```

Comment:
This seems suspect here. You've already used `start` in the calculation of `end` above, and if you change `start` now, then you're not taking into account how that would change `end`. Also, passing `start` past the end of the string causes the substring to be taken from the *beginning* of the string? A better outcome would be an empty string, consistent with what happens when passing `end` past the end of the string.

Reply:
I'm not sure what you're suggesting I do here.  If start is <= 0, or greater than the length of the string, I should just return a blank string?

Reply:
It seems to me that certainly if start is past the end of the string, you should return an empty string. If the start less than zero, it's up to you. You could return an error, or you could adjust things to return only the part of the (start of the string) that is covered by the start and count.